### ISSUE TITLE
Docs build failing on readthedocs

### ISSUE THREAD

**Issue by lucabaldini** [lucabaldini/ixpeobssim#683]
This happened between versions 30.0.0 and 30.0.1, but I don't think it has anything to do with the small fix going into the latter---it must be something having to do with the readthedocs environment.

```
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/cmd/build.py", line 284, in build_main
    app.build(args.force_all, args.filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/application.py", line 347, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 311, in build_update
    self.build(to_build,
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 327, in build
    updated_docnames = set(self.read())
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 434, in read
    self._read_serial(docnames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 455, in _read_serial
    self.read_doc(docname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 511, in read_doc
    publisher.publish()
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/core.py", line 224, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/io.py", line 105, in read
    self.parse()
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/readers/__init__.py", line 76, in parse
    self.parser.parse(self.input, document)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/parsers.py", line 80, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 169, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2785, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 325, in section
    self.new_subsection(title, lineno, messages)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 391, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 279, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2799, in text
    paragraph, literalnext = self.paragraph(lines, startline)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 416, in paragraph
    textnodes, messages = self.inline_text(text, lineno)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 425, in inline_text
    nodes, messages = self.inliner.parse(text, lineno,
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 649, in parse
    before, inlines, remaining, sysmessages = method(self, match,
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 792, in interpreted_or_phrase_ref
    nodelist, messages = self.interpreted(rawsource, escaped, role,
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 889, in interpreted
    nodes, messages2 = role_fn(role, rawsource, text, lineno, self)
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/ext/extlinks.py", line 103, in role
    title = caption % part
TypeError: not all arguments converted during string formatting

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/ixpeobssim/envs/latest/lib/python3.9/site-packages/sphinx/ext/extlinks.py", line 103, in role
    title = caption % part
TypeError: not all arguments converted during string formatting
The full traceback has been saved in /tmp/sphinx-err-khy0f6xu.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!

```

**Comment by lucabaldini** [comment]
I think this is related to https://gitlab.haskell.org/ghc/ghc/-/merge_requests/9625

**Comment by lucabaldini** [comment]
I deployed a fix that I am testing on readthedocs right now, see https://github.com/lucabaldini/ixpeobssim/pull/690

**Comment by lucabaldini** [comment]
Shipped in 30.2.1
